flocker/acceptance/test_api.py

```diff
@@ -295,10 +310,26 @@ def get_test_cluster(test_case, node_count):
                        "{existing} node(s) are set.".format(
                            necessary=node_count, existing=len(agent_nodes)))
 
-    return succeed(Cluster(
+    cluster = Cluster(
         control_node=Node(address=control_node),
-        nodes=map(lambda address: Node(address=address), agent_nodes),
-    ))
+        nodes=[]
+    )
+
+    # Wait until nodes are up and running:
+    def nodes_available():
+        d = cluster.current_nodes()
+        d.addCallback(lambda (cluster, nodes): len(nodes) == node_count)
+        return d
+    agents_connected = loop_until(nodes_available)
```

Comment:
:question: A lot of the hassle we have now with the acceptance tests comes from the lack of logging, particularly in loops without their own timeout / exception raising - a failed test is often just a `TimeoutError`. It might be worth adding a log message here. Optional because the issue mentions that this will change soon but do it if it is quick.